checkpointManager.py

```diff
@@ -1,9 +1,142 @@
 from checkpoints import *
-
+from scenarioManager import *
 class CheckpointManager:
-    def __init__(self, checkpoints):
+    
+    def __init__(self, checkpoints,scenarioManager,player):
         self.checkpoints = checkpoints
+        #to track where we're at, the state of the player within the checkpoints
+        self.currentCheckpoint = checkpoints[0]
+        self.scenarioManager = scenarioManager
+        self.player = player
+
+        #cost of ferry, Merchant pays .9* as much, scenarios currently use magic numbers for the cost
+        self.ferryCost=150
+        #initialize the player distance to next
+        player.distNext=self.currentCheckpoint.distToNextCP
     
-    def updateState(self):
+    #checkpoints are linear, so this will advance one checkpoint forward. if the checkpoint manager is already at the last checkpoint, call the gameover scenario
+    def nextCheckpoint(self):
         print("Updating checkpoint manager state...")
-        # Implement the logic to update the state of the checkpoint manager here
\ No newline at end of file
+        #if the player has no distance left to the checkpoint, advance to the next checkpoint
+        if(self.player.distNext <= 0):
+            indexOfCurrentCheckpoint = self.checkpoints.index(self.currentCheckpoint)
+            indexOfCurrentCheckpoint +=1
+            if(indexOfCurrentCheckpoint == len(self.checkpoints)-1):
+                #final checkpoint, game is over
+                print("Game over!")
+                self.scenarioManager.callScenarioByName("Game Over")
+            else:
+                self.currentCheckpoint=self.checkpoints[indexOfCurrentCheckpoint]
+                #player disToNextCP can be updated
+                self.player.distNext=self.currentCheckpoint.distToNextCP
+                #call the river() or town() menus once arriving at a new checkpoint
+                #once there the gui is ready to display choices these wont need to do a terminal
+                if self.currentCheckpoint.isRiver:
+                    self.riverScenario()
+                else:
+                    self.townScenario()
+                
+        else:
+            #player still has distance to go, call the player. travel checkpoint
+            self.scenarioManager.callScenarioByName("Travel")
+            if(self.player.distNext <= 0):
+                #if the palyer hits 0 distance to the next checkpoint, this will call nextCheckpoint again to pull up river or town scenarios
+                self.nextCheckpoint()
+
+        
+            
+    #if default choice of -1 is used, use terminal to poll choices
+    #if choice isn't -1, then the input comes from a GUI selection and thus a specific scenario can be called directly from the scenario manager
+    def riverScenario(self,choice = -1):
```

Comment:
The entire river scenario should be its own scenario. It should not be divided into two.
Please move this code into the "scenarioRiver" file